packages/misc/src/lib/misc.ts

```diff
@@ -691,3 +694,115 @@ export function getEnv({
   // Default
   return defaultValue;
 }
+
+export function sendRequest(
+  url: string,
+  req: RequestInit,
+  requestId: string
+): Promise<Response> {
+  return fetch(url, req)
+    .then(async (response) => {
+      const isJson = response.headers
+        .get('content-type')
+        ?.includes('application/json');
+
+      const data = isJson ? await response.json() : null;
+
+      if (!response.ok) {
+        // get error message from body or default to response status
+        const error = data || response.status;
+        return Promise.reject(error);
+      }
+
+      return data;
+    })
+    .catch((error: NodeErrorV3) => {
+      logErrorWithRequestId(
+        requestId,
+        `Something went wrong, internal id for request: lit_${requestId}. Please provide this identifier with any support requests. ${
+          error?.message || error?.details
+            ? `Error is ${error.message} - ${error.details}`
+            : ''
+        }`
+      );
+      return Promise.reject(error);
+    });
+}
+
+/**
+ * Allows for invoking a callback and re exucting while re generating a new request identifier
+ * @param execCallback
+ * @param errorCallback
+ * @param opts
+ * @returns {T}
+ */
+export async function executeWithRetry<T>(
+  execCallback: (requestId: string) => Promise<T>,
+  errorCallback?: (error: any, requestId: string, isFinal: boolean) => void,
+  opts?: any
```

Comment:
We can use `RetryTollerance` here for `opts`.

Reply:
🤦 totally correct. was a bit sleepy when doing the last refactor. Good Catch!